### PR TITLE
Feat/make gui code more c++ friendly zappy gui

### PR DESCRIPTION
**Pull request by Thomaltarix**
I made the GUI code more c++ friendly.
In fact, i made more inheritance and abstractions of existing classes for the GUI.
With this changes, we can build more test, for the Network, the updater, etc..

To see more details, check this issue: #181

### CODE REVIEW

File: gui/include/GUIUpdater/AGUIUpdater.hpp [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+** Zappy
+** File description:
+** AGUIUpdater
+*/
+
+#pragma once
+
+#include "GameDatas/GameData.hpp"
+#include "Network/Network.hpp"
+#include "GUIUpdater/IGUIUpdater.hpp"
```
Comment (mathieurobert1) [inline gui/include/GUIUpdater/AGUIUpdater.hpp]:
```suggestion
#include "Network/Network.hpp"
#include "GameDatas/GameData.hpp"
#include "GUIUpdater/IGUIUpdater.hpp"
```

[the remark this answers is not in the document]
Comment (Thomaltarix) [reply gui/include/GUIUpdater/AGUIUpdater.hpp:12]: Done

File: gui/src/Engine/Engine.cpp [hunk 2]
```diff
@@ -5,27 +5,29 @@
 ** Engine
 */
 
-#include "Event/Event.hpp"
 #include "Engine/Engine.hpp"
-#include "GUIUpdater/GUIUpdater.hpp"
 #include "Colors.hpp"
```
Comment (mathieurobert1) [inline gui/src/Engine/Engine.cpp]: Inverse includes.

File: gui/include/Parsing/IServerParser.hpp [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        /**
+         * @brief Destroy the IServerParser object.
+         */
+        virtual ~IServerParser() = default;
+
+        /**
+         * @brief Parse the command server.
+         *
+         * @param command Command to parse.
+         * @return std::vector<std::string> - arguments parsed
```
[the remark this answers is not in the document]
Comment (Thomaltarix) [reply gui/include/Parsing/IServerParser.hpp:34]: Done

File: gui/include/Parsing/ServerParser.hpp [hunk 4]
```diff
@@ -57,8 +59,6 @@ class Gui::ServerParser {
             LIST_INT
         };
 
-    private:
-
         /**
          * @brief Map of types to parse related to the command
```
[the remark this answers is not in the document]
Comment (Thomaltarix) [reply gui/include/Parsing/ServerParser.hpp:63]: Done

File: gui/src/Engine/Engine.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (Thomaltarix) [reply gui/src/Engine/Engine.cpp:9]: Done

**Review by mathieurobert1 — CHANGES_REQUESTED** [review]
Just includes to inverse.

**Review by AxelF44 — APPROVED** [review]

**Review by mathieurobert1 — APPROVED** [review]

**Pull request merged**